sql/core/src/test/scala/org/apache/spark/sql/internal/ExecutorSideSQLConfSuite.scala

```diff
@@ -125,6 +125,38 @@ class ExecutorSideSQLConfSuite extends SparkFunSuite with SQLTestUtils {
     val e = intercept[SparkException](dummyQueryExecution1.toRdd.collect())
     assert(e.getCause.isInstanceOf[NoSuchElementException])
   }
+
+  test("SPARK-30556 propagate local properties to subquery execution thread") {
+    withSQLConf("spark.sql.subquery.maxThreadThreshold" -> "1") {
```

Comment:
Shall we use `SUBQUERY_MAX_THREAD_THRESHOLD` instead of `"spark.sql.subquery.maxThreadThreshold"`? That's better when we change the configuration string name later.

Reply:
updated